.eslintrc

```diff
@@ -1,13 +1,14 @@
 {
   "parserOptions": {
-    "ecmaVersion": 6,
-    "sourceType": "module"
+    "ecmaVersion": 2018, // suppport syntax features (like Object spread operator) up to es2018
+    "sourceType": "module",
+    "ecmaFeatures": {
+      "impliedStrict": true
+    }
   },
   "env": {
+    "es2017": true, // suppport globals (like Promise) up to es2017 (es2018 doesn't have new globals)
```

Comment:
Shouldn't it match `ecmaVersion`?

Reply:
Actually, env = globals (like Promise), parserOptions.ecmaVersion = syntax features (like Object spread operator).
ES2018 doesn't have any new globals, so ESlint doesn’t allow to set env=2018: https://eslint.org/docs/user-guide/configuring/language-options#specifying-environments